miden/src/cli/prove.rs

```diff
@@ -57,12 +57,13 @@ impl ProveCmd {
         let now = Instant::now();
 
         // fetch the stack and program inputs from the arguments
-        let program_inputs = input_data.get_program_inputs()?;
         let stack_inputs = input_data.get_stack_inputs()?;
+        let program_inputs = input_data.get_program_inputs()?;
```

Comment:
minor nit: Maybe `get_program_inputs()` can be changed to just `program_inputs()` and similarly for `get_stack_inputs()`? (I know it's not a part of this PR)

Reply:
We parse these on the fly instead of doing only once during the instance creation.

I think we have a couple of things to rework on this function, being the main the inability to serialize/deserialize generic advice providers - and when deserialized, have a concrete instance parsed directly from the input, instead of having a string that will parse on the fly.

Reply:
Makes sense